### PR TITLE
Parallel downloading

### PR DESCRIPTION
**Pull request by casz**
Hopefully, this should speed up downloads 😅

time diff I did on our plugins.txt:
.jar after:
```
real    2m 55.19s
user    0m 10.79s
sys     0m 3.04s
```
.jar before:
```
real    8m 1.32s
user    0m 15.58s
sys     0m 5.06s
```
.sh:
```
real    0m 48.37s
user    0m 7.77s
sys     0m 4.02s
```

### CODE REVIEW

File: plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java [hunk 1]
```diff
@@ -194,13 +194,13 @@ public void outputFailedPlugins() {
      * @param plugins list of plugins to download
      */
     public void downloadPlugins(List<Plugin> plugins) {
-        for (Plugin plugin : plugins) {
+        plugins.parallelStream().forEach(plugin -> {
```
Comment (timja) [inline plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java]: can you run it on your setup and see if it helps 😄

Comment (casz) [reply plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java]: That was the plan! I am just abusing Jenkins CI to get the jar :)

Comment (casz) [reply plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java]: Holy crap 😓
```
real    1m 6.93s
user    0m 8.98s
sys     0m 1.86s
```

I think we can optimize it even more if we play with custom thread pool and just doing it on all available cores (+hyper threading)

Comment (casz) [reply plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java]: but reading up on optimizing I would say NOPE 😅

**Comment by casz** [comment]
Okay after fixing #50 the full download takes
```
real    2m 55.19s
user    0m 10.79s
sys     0m 3.04s
```
Still a huge improvement but I think we can do better!

**Comment by casz** [comment]
I'd say let's merge this at least and then we can worry about improving the download even more!

**Pull request merged**